service/src/main/kotlin/fi/espoo/evaka/application/PlacementToolService.kt

```diff
@@ -70,68 +73,52 @@ class PlacementToolService(
-                                nameSv = it.nameSv,
-                                nameEn = it.nameEn,
-                                validPlacementType = it.validPlacementType,
-                            )
-                        }
-                if (null == partTimeServiceNeedOption) {
+                        ?.id
+                if (null == partTimeServiceNeedOptionId) {
                     throw NotFound("No part time service need option found")
                 }
-                val defaultServiceNeedOption =
+                val defaultServiceNeedOptionId =
                     serviceNeedOptions
                         .firstOrNull {
                             it.validPlacementType == PlacementType.PRESCHOOL_DAYCARE &&
                                 it.defaultOption
                         }
-                        ?.let {
-                            ServiceNeedOption(
-                                id = it.id,
-                                nameFi = it.nameFi,
-                                nameSv = it.nameSv,
-                                nameEn = it.nameEn,
-                                validPlacementType = it.validPlacementType,
-                            )
-                        }
-                if (null == defaultServiceNeedOption) {
+                        ?.id
+                if (null == defaultServiceNeedOptionId) {
                     throw NotFound("No default service need option found")
                 }
-                val nextPreschoolTerm =
-                    tx.getPreschoolTerms().firstOrNull { it.finnishPreschool.start > clock.today() }
-                if (null == nextPreschoolTerm) {
+                val nextPreschoolTermId =
+                    tx.getPreschoolTerms()
+                        .firstOrNull { it.finnishPreschool.start > clock.today() }
+                        ?.id
+                if (null == nextPreschoolTermId) {
                     throw NotFound("No next preschool term found")
                 }
-                val placements = parsePlacementToolCsv(file.inputStream)
-                placements
-                    .forEach { data ->
-                        asyncJobRunner.plan(
-                            tx,
-                            listOf(
-                                AsyncJob.PlacementTool(
-                                    user,
-                                    data,
-                                    partTimeServiceNeedOption,
-                                    defaultServiceNeedOption,
-                                    nextPreschoolTerm,
-                                )
-                            ),
-                            runAt = clock.now(),
-                            retryCount = 1,
-                        )
-                    }
+                val placements: List<PlacementToolData>
+                file.inputStream.use { placements = parsePlacementToolCsv(it) }
```

Comment:
:+1:

Vinkki: `.use` välittää myös paluuarvon joten tässä pääsisi vielä helpommalla näin:

```kotlin
val placements = file.inputStream.use { parsePlacementToolCsv(it) }
```